core/include/detray/propagator/navigator.hpp

```diff
@@ -344,6 +344,14 @@ class navigator {
             return m_status == navigation::status::e_on_portal;
         }
 
+        /// Helper method to check the track has encountered material
+        DETRAY_HOST_DEVICE
+        inline auto encountered_material() const -> bool {
+            return (is_on_module() or is_on_portal()) and
```

Comment:
why is `is_on_portal()` considered here?

Reply:
Because most material maps will reside on portals